google-cloud-logging/src/main/java/com/google/cloud/logging/LoggingImpl.java

```diff
@@ -792,6 +792,50 @@ private static LogName getLogName(
     return destination.toLogName(logName);
   }
 
+  public Iterable<LogEntry> populateMetadata(
+      Iterable<LogEntry> logEntries,
+      MonitoredResource customResource,
+      String... exclusionClassPaths) {
+    final Boolean needDebugInfo =
+        Iterables.any(
+            logEntries,
+            log -> log.getSeverity() == Severity.DEBUG && log.getSourceLocation() == null);
+    final SourceLocation sourceLocation =
+        needDebugInfo ? SourceLocation.fromCurrentContext(exclusionClassPaths) : null;
+    // populate monitored resource metadata by prioritizing the one set via
+    // WriteOption
+    final MonitoredResource resourceMetadata =
+        customResource == null
+            ? MonitoredResourceUtil.getResource(getOptions().getProjectId(), null)
+            : customResource;
+    final Context context = (new ContextHandler()).getCurrentContext();
+    final ArrayList<LogEntry> populatedLogEntries = Lists.newArrayList();
+
+    // populate empty metadata fields of log entries before calling write API
+    for (LogEntry entry : logEntries) {
+      LogEntry.Builder builder = entry.toBuilder();
```

Comment:
Can `entry` be null here?

Reply:
In theory it can. I will add a check, thanx.